run_service.py

```diff
@@ -0,0 +1,228 @@
+                    }
+                ),
+            }
+        ),
+    }
+)
+
+
+def print_box(text: str, margin: int = 1, character: str = '=') -> None:
+    """Print text centered within a box."""
+
+    lines = text.split('\n')
+    text_length = max(len(line) for line in lines)
+    length = text_length + 2 * margin
+
+    border = character * length
+    margin_str = ' ' * margin
+
+    print(border)
+    print(f"{margin_str}{text}{margin_str}")
+    print(border)
+    print()
+
+
+def print_title(text: str) -> None:
+    """Print title."""
+    print()
+    print_box(text, 4, '=')
+
+
+def print_section(text: str) -> None:
+    """Print section."""
+    print_box(text, 1, '-')
+
+
+def wei_to_unit(wei: int) -> float:
+    """Convert Wei to unit."""
+    return wei / 1e18
+
+
+def wei_to_token(wei: int, token: str = "xDAI") -> str:
+    """Convert Wei to token."""
+    return f"{wei_to_unit(wei):.2f} {token}"
+
+
+def ask_confirm_password() -> str:
+    password = getpass.getpass("Please enter a password: ")
+    confirm_password = getpass.getpass("Please confirm your password: ")
+
+    if password == confirm_password:
+        return password
+    else:
+        print("Passwords do not match. Terminating.")
+        sys.exit(1)
+
+
+def check_rpc(rpc_url: str) -> None:
+    spinner = Halo(text=f"Checking RPC...", spinner="dots")
+    spinner.start()
+
+    rpc_data = {
+        "jsonrpc": "2.0",
+        "method": "eth_newFilter",
+        "params": ["invalid"],
+        "id": 1
+    }
+
+    try:
+        response = requests.post(
+            rpc_url,
+            json=rpc_data,
+            headers={"Content-Type": "application/json"}
+        )
+        response.raise_for_status()
+        rpc_response = response.json()
+    except Exception as e:
+        print("Error: Failed to send RPC request:", e)
+        sys.exit(1)
+
+    rcp_error_message = rpc_response.get("error", {}).get("message", "Exception processing RCP response")
+
+    if rcp_error_message == "Exception processing RCP response":
+        print("Error: The received RCP response is malformed. Please verify the RPC address and/or RCP behavior.")
+        print("  Received response:")
+        print("  ", rpc_response)
+        print("")
+        print("Terminating script.")
+        sys.exit(1)
+    elif rcp_error_message == "Out of requests":
+        print("Error: The provided RCP is out of requests.")
+        print("Terminating script.")
+        sys.exit(1)
+    elif rcp_error_message == "The method eth_newFilter does not exist/is not available":
+        print("Error: The provided RPC does not support 'eth_newFilter'.")
+        print("Terminating script.")
+        sys.exit(1)
+    
+    spinner.succeed("RPC checks passed.")
+
+
+def main() -> None:
+    """Run service."""
+
+    print_title("Trader Quickstart")
+    print("This script will assist you in setting up and running the Trader service.")
```

Comment:
Shouldnt the `Trader` word be taken from the Configuration?

Reply:
Thanks for your comments @8ball030. This was an experimental PR to use the middleware in this repository, but at that time the middleware lacked some required features. For this reason, this PR is somewhat obsolete. The middleware has evolved and will be eventually integrated in this repository, most likely in a new PR.